src/sparseml/pytorch/sparsification/pruning/modifier_pruning_mfac.py

```diff
@@ -130,6 +132,8 @@ class MFACPruningModifier(BaseGradualPruningModifier):
     :param mask_type: String to define type of sparsity to apply. May be 'unstructred'
         for unstructured pruning or 'block4' for four block pruning or a list of two
         integers for a custom block shape. Default is 'unstructured'
+    :param grad_sampler_kwargs: kwargs to override default train dataloader config
```

Comment:
It's very confusing to tell the difference between `grad_sampler_kwargs` and `grad_sampler`, since `grad_sampler_kwargs` is a list of kwargs for to pass to the data loader builder and `grad_sampler` is a list of kwargs that are used to build the data loader and a loss fn pass to the `GradSampler()`. If it has to stay this way, I propose renaming

`grad_sampler_kwargs` -> `data_loader_kwargs` or `grad_sampler_data_loader_kwargs`
`grad_sampler` -> `grad_sampler_kwargs`

Otherwise, I'd make the argument `grad_sampler_kwargs` which contains a key for the data loader, the data loader's arguments, and the loss fn

Reply:
@kylesayrs thanks for the review and useful comments!
In the initial versions of MFAC pruner class these gradient sampling kwargs were divided in two groups: one for dataloader and one for loss function. But there was a pushback as it was considered to be too complicated for end users to configure (don't know if these views have changed now). 

I agree that the naming should be improved, but I would try to keep `grad_sampler_kwargs` in the yaml recipe as changing it would require updating all existing recipes from oBERT runs. I would rather try to adopt a different naming for `grad_sampler = {"data_loader_builder": ..., "loss_function": ...}` dictionary. What do you think?

Reply:
agreed on keeping the naming as is, especially given relationship between the recipe args, name and actual python object